src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -20,18 +24,36 @@ public void Initialize(TelemetryConfiguration configuration)
 
             _listener = new ActivityListener
             {
-                ShouldListenTo = source => source.Name.StartsWith("Microsoft.Azure.Functions.Worker"),
+                ShouldListenTo = source => source.Name.StartsWith(TraceConstants.FunctionsActivitySource),
                 ActivityStarted = activity =>
                 {
                     var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(activity);
```

Comment:
will this create a `RequestTelemetry`?

Reply:
No -- `DependencyTelemetry`.

Although I guess this is a good question for App Insights experts... this is technically a distributed call across processes. Should we treat this as a separate service and create a `RequestTelemetry` with a parent of the host's `RequestTelemetry`?

Any opinions @lmolkova?

Reply:
what's the appropriate Activity.Kind in this case? If we create an activity with Server/Consumer, Az Mon exporter translates that to a RequestTelemetry. We definitely have clinet and server relationship between host and worker. Another thing that we have to check is how it looks in the Application Map.

Reply:
Interesting question. Apparently, Java workers don't report any telemetry about worker invocation fully relying on host trigger `RequestTelemetry` and faking parent on the nested calls. It might be possible in .NET, but would be hacky. 

Since this package is in preview, I would not block on that, especially if you want users to have a similar experience on OTel.

Then the next question is how to make it look best in AI:
- host will record RequestTelemetry
- worker would need to report DependencyTelemetry (as a child of request). Adding another request would mess up AI UX. To make it work properly, the type of this telemetry should be "InProc". Otherwise AppMap will be broken among other things. 
- there will be child auto-traced activities nested under worker invocation dependency.

Internal activity maps to InProc dependency telemetry - we use in Azure SDKs for the same reasons to tell UX where is the real network call and where is a logical operation.

Regarding Activity.Kind - It seems faas otel spec only talks about outer (host) request, but afaik AWS lambda reports several spans too - it'd be nice to clarify with otel faas WG and write down in the spec.

Reply:
This is awesome information, thanks @lmolkova! It does seem like we want this tracked as an "InProc" Dependency for the best App Insights experience.